src/libtools/examples/merging.cpp

```diff
@@ -0,0 +1,173 @@
+#include <kdb.hpp>
+#include <keysetio.hpp>
+
+#include <iostream>
+#include <string>
+
+#include <merging/threewaymerge.hpp>
+#include <merging/metamergestrategy.hpp>
+#include <merging/automergestrategy.hpp>
+#include <merging/onesidestrategy.hpp>
+
+
+using namespace kdb;
+using namespace kdb::tools::merging;
+using namespace std;
+
+int main()
+{
+	KeySet ours;
+	KeySet theirs;
+	KeySet base;
+
+	// the root of the subtree containing our keys (i.e. our side of the merge)
+	Key oursRoot ("user/ours", KEY_END);
+
+	// the root of the subtree containing their keys (i.e. their side of the merge)
+	Key theirsRoot ("user/theirs", KEY_END);
+
+	// the root of the subtree containing the base keys (i.e. the common ancestor of the merge)
+	Key baseRoot ("user/base", KEY_END);
+
+	// the root of the subtree that will contain the merge result
+	Key resultRoot ("user/result", KEY_END);
+
+	// Step 1: retrieve clean KeySets containing only those
```

Comment:
Please link to the example of kdbSet(): its another important use case of 3 way merging (merging when kdbSet() reports an conflict). In that case one KDB instance is enough.

http://doc.libelektra.org/api/latest/html/group__kdb.html#ga11436b058408f83d303ca5e996832bcf